### PR TITLE
reference: update transaction related master docs

### PR DESCRIPTION
**Pull request by cfzjywxk**
### What is changed, added or deleted? (Required)

Update some error code changes and transaction-related documents.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v4.0 changes for now)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: reference/transactions/transaction-pessimistic.md [hunk 1]
```diff
@@ -55,6 +55,7 @@ category: reference
 - 通过设置 `innodb_lock_wait_timeout` 变量，设置等锁超时时间，等锁超时后返回兼容 MySQL 的错误码 `1205`。
 
 - 支持 `FOR UPDATE NOWAIT` 语法，遇到锁时不会阻塞等锁，而是返回兼容 MySQL 的错误码 `3572`。
+- Point Get 和 Batch Point Get 算子，如果没有读到数据，依然会对给定的主键或者唯一键加锁，阻塞其他事务对于相同主键唯一键加锁或者写入操作。
```
Comment (TomShawn) [inline reference/transactions/transaction-pessimistic.md:58]:
```suggestion
- 如果 `Point Get` 和 `Batch Point Get` 算子没有读到数据，依然会对给定的主键或者唯一键加锁，阻塞其他事务对相同主键唯一键加锁或者进行写入操作。
```

File: reference/transactions/transaction-pessimistic.md [hunk 2]
```diff
@@ -68,7 +69,7 @@ category: reference
 
 3. DDL 可能会导致悲观事务提交失败。
 
-    MySQL 在执行 DDL 时会被正在执行的事务阻塞住，而在 TiDB 中 DDL 操作会成功，造成悲观事务提交失败：`ERROR 1105 (HY000): Information schema is changed. [try again later]`。
+    MySQL 在执行 DDL 时会被正在执行的事务阻塞住，而在 TiDB 中 DDL 操作会成功，造成悲观事务提交失败：`ERROR 1105 (HY000): Information schema is changed. [try again later]`。TiDB 事务执行过程中并发执行 TRUNCATE TABLE 语句，可能会导致事务报错 `table dosen't exist`。
```
Comment (TomShawn) [inline reference/transactions/transaction-pessimistic.md:72]:
```suggestion
    MySQL 在执行 DDL 时会被正在执行的事务阻塞住，而在 TiDB 中 DDL 操作会成功，造成悲观事务提交失败：`ERROR 1105 (HY000): Information schema is changed. [try again later]`。TiDB 事务执行过程中并发执行 `TRUNCATE TABLE` 语句，可能会导致事务报错 `table dosen't exist`。
```

File: reference/transactions/transaction-pessimistic.md [hunk 3]
```diff
@@ -77,6 +78,7 @@ category: reference
     所有自动提交的语句都不会加悲观锁，该类语句在用户侧感知不到区别，因为悲观事务的本质是把整个事务的重试变成了单个 DML 的重试，autocommit 事务即使在 TiDB 关闭重试时也会自动重试，效果和悲观事务相同。
 
     自动提交的 select for update 语句也不会等锁。
+6. 不会对于语句中包含的 EMBEDDED SELECT 读到的相关数据加锁。
```
Comment (TomShawn) [inline reference/transactions/transaction-pessimistic.md:81]:
```suggestion

6. 对语句中 `EMBEDDED SELECT` 读到的相关数据不会加锁。    
```

File: reference/transactions/transaction-pessimistic.md [hunk 4]
```diff
@@ -77,6 +78,7 @@ category: reference
     所有自动提交的语句都不会加悲观锁，该类语句在用户侧感知不到区别，因为悲观事务的本质是把整个事务的重试变成了单个 DML 的重试，autocommit 事务即使在 TiDB 关闭重试时也会自动重试，效果和悲观事务相同。
 
     自动提交的 select for update 语句也不会等锁。
+6. 对语句中 `EMBEDDED SELECT` 读到的相关数据不会加锁。
```
Comment (ran-huang) [inline reference/transactions/transaction-pessimistic.md:81]: We should leave a blank line before a list.

**Pull request merged**